streams/src/main/java/org/apache/kafka/streams/processor/internals/AbstractTask.java

```diff
@@ -172,7 +174,14 @@ public String toString(final String indent) {
      * Flush all state stores owned by this task
      */
     void flushState() {
-        stateMgr.flush();
+        try {
+            stateMgr.flush();
+        } catch (final ProcessorStateException e) {
+            if (e.getCause() instanceof ProducerFencedException) {
+                log.warn("Caught a recoverable Kafka exception while flushing state. Initiating a rebalance to attempt recovery.", e);
```

Comment:
nit: e -> e.getCause()

Reply:
Thanks! I didn't unwrap it in the log, since this preserves the full call stack (which helps track down why the exception happened)

Reply:
I don't think we should log anything here -- we log and info level statement when we _catch_ a `TaskMigratedException` and it's contains the full context as we pass in `this` and `e` -- otherwise, we would double log (also WARN does not seem to be appropriate anyway)

Reply:
Roger that. I can remove it.